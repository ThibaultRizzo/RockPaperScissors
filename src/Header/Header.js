import React, { Component } from 'react';

class Header extends Component {
  render() {
    const title = 'Waste an Hour Having Fun';
    return (
      <div id="Header">
        <header>{title}</header>
        <div id="header-options">
          <input
            type="button"
            value="New Game"
            className={
              'special-button ' + (this.props.isGameFinished ? 'jump' : '')
            }
            onClick={e => this.props.onCreateNewGame(null)}
          />
        </div>
      </div>
    );
  }
}

export default Header;
