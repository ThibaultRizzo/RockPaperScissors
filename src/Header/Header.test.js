import React from 'react';
import ReactDOM from 'react-dom';
import Header from './Header';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const startState = { isGameFinished: false, setGameType: jest.fn() };
  const rendered = renderer.create(
    <Header
      isGameFinished={startState.isGameFinished}
      onCreateNewGame={startState.setGameType}
    />
  );
  expect(rendered.toJSON()).toMatchSnapshot();
});
