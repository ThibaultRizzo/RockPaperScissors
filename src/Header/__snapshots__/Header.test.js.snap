// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  id="Header"
>
  <header>
    Waste an Hour Having Fun
  </header>
  <div
    id="header-options"
  >
    <input
      className="special-button "
      onClick={[Function]}
      type="button"
      value="New Game"
    />
  </div>
</div>
`;
